[hotfix] docker runner fixes

arnout opened the pull request:
Various small fixes to make the runner image work on my laptop.

Note that the entrypoint update requires an update of the docker images on dockerhub.
However, since nobody else seems to be bothered by this, there's no urgency for doing the update.

arnout committed:
tools: run.sh: create log directory

podman (a docker replacement) doesn't automatically create the host
directory of a --volume argument - it fails when it doesn't exist.

Therefore, create the log directory which gets mapped to
/tmp/beerocks/logs before starting the container.

Signed-off-by: Arnout Vandecappelle (Essensium/Mind) <arnout@mind.be>

arnout committed:
docker: runner: improve setting of IP address of bridge

When docker is configured for IPv6, eth0 gets both an IPv4 and an IPv6
address. Both of them match the 'inet' grep expression, so it yields an
invalid address (a long one that contains both).

Improve this by properly anchoring the regular expression.

While we're at it, move the regex match into the awk command instead of
combining grep and awk.

Signed-off-by: Arnout Vandecappelle (Essensium/Mind) <arnout@mind.be>

arnout committed:
docker: runner: save bridge MAC address

Due to limitations of the controller topology database (each node must
have a different MAC address), the bridge MAC must be different from any
interface MAC.

Apparently, this doesn't cause an issue in CI at the moment. Probably
that's because the fake ETHERNET node that is created also has the same
IP address as the backhaul interface address, and these problems
counteract each other. However, this accidental solution only happens
because docker generates sequential MAC addresses for the interfaces. If
random addresses are used instead (which is what podman does), things
break down again.

Fix this by setting the br-lan MAC address explicitly instead of
inheriting the address of one of the interfaces. When the an interface
is added to the bridge, the bridge inherits its MAC address. So save the
(random) MAC address that we get when the bridge is created, and
re-apply it after adding all interfaces.

Signed-off-by: Arnout Vandecappelle (Essensium/Mind) <arnout@mind.be>

rmelotte commented on tools/docker/runner/entrypoint.sh:
```diff
@@ -24,6 +25,7 @@ run ip link set      dev wlan2    master br-lan
 run ip address flush dev eth0
 run ip link set      dev wlan0    up
 run ip link set      dev wlan2    up
+run ip link set      dev br-lan   addr "$bridge_mac"
```
For the Turris Omnia, we have this in the wiki:
```
The backhaul interface must have a different MAC address than the bridge interface. 
```

Does it not apply to the dummy variant?

arnout replied:
That is an extremely good observation! I missed the fact that originally we had a fixed MAC address that was different from eth0.

After you removed the explicit setting of the addresses, br-lan receives the same address as eth0 in CI:
```
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN mode DEFAULT group default qlen 1
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
2: br-lan: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP mode DEFAULT group default qlen 1000
    link/ether 02:42:ac:13:00:03 brd ff:ff:ff:ff:ff:ff
3: wlan0: <BROADCAST,NOARP,UP,LOWER_UP> mtu 1500 qdisc noqueue master br-lan state UNKNOWN mode DEFAULT group default qlen 1000
    link/ether d6:82:42:41:62:97 brd ff:ff:ff:ff:ff:ff
4: wlan2: <BROADCAST,NOARP,UP,LOWER_UP> mtu 1500 qdisc noqueue master br-lan state UNKNOWN mode DEFAULT group default qlen 1000
    link/ether c2:39:82:4b:44:b1 brd ff:ff:ff:ff:ff:ff
188015: eth0@if188016: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master br-lan state UP mode DEFAULT group default 
    link/ether 02:42:ac:13:00:03 brd ff:ff:ff:ff:ff:ff link-netnsid 0
```

Apparently, it now works by accident... I think it might have something to do with the way that docker generates MAC addresses: they are sequential, so repeater1's address is the gateway address + 1. This happens to be the same as the fake address that is generated for gateway's ETHERNET node, and apparently the two problems compensate for each other...

But this might very well explain why test_flows still doesn't work for me. I'll look further into it.

rmelotte reviewed (approved):
LGTM.
Why is it ok in this case to use the same MAC address for both the bridge and eth0 though?

arnout committed:
docker: run.sh: remove unused function gateway_netid_length()

The only caller of the function was removed in c60d147d12aa, so remove
the function definition as well.

Signed-off-by: Arnout Vandecappelle (Essensium/Mind) <arnout@mind.be>

rmelotte commented on tools/docker/runner/entrypoint.sh:
```diff
@@ -11,13 +11,24 @@ run() {
     "$@" || exit $?
```
The commit message says "we could use any of the techniques
in [1]" but there is no "[1]".
Did you forget to add it?

rmelotte reviewed (approved):
Weird, when I add a bridge interface interactively on my laptop I get different MAC addresses...

abelog reviewed (approved):
LGTM

arnout commented:
> Weird, when I add a bridge interface interactively on my laptop I get different MAC addresses...

Indeed, when I do it inside the container, `ip link set      dev eth0     master br-lan` makes br-lan get the MAC address of eth0. But when I do it directly on my laptop, this doesn't happen...

However, this makes me realize that br-lan already has a random address, so I don't need to create the dummy interface! Updating coming in...
